### PR TITLE
Apply manual mean baseline to the response-locked epochs 

### PR DESCRIPTION
**Pull request by mscheltienne**
@ArthurNguyen If you want to convince yourself that this is correct, here is an example of the same baseline code with toy-data:

```
from copy import deepcopy

import numpy as np
from matplotlib import pyplot as plt
from mne import create_info
from mne._fiff.pick import _picks_to_idx

# create mock dataset
sfreq = 1024
n_channels = 3  # 2 eeg, 1 misc
info = create_info(["EEG1", "EEG2", "MISC1"], sfreq, ch_types=["eeg", "eeg", "misc"])
events = np.array([400, 2000, 4000, 6300])
events_rp = np.array([610, 4180])
data = np.zeros((n_channels, 10000), dtype=float)
data_rp = np.zeros((events_rp.size, 2, 500), dtype=float)  # 2 eeg channels
# first epoch
data[0, 200:400] = 1.0
data[1, 200:400] = 0.5
data[:, 610:910] = 5.0
data[0, 910:1110] = 4.0
data[1, 910:1110] = 3.0
data_rp[0, :, :] = data[:2, 610:1110]
# second epoch
data[:, 3800:4000] = 2.0
data[:, 4180:4680] = 6
data_rp[1, :, :] = data[:2, 4180:4680]

# apply baseline correction
distance = np.full(
    (events.size, events_rp.size),
    events_rp,
    dtype=float,
)
distance = distance.T - events
distance[np.where(distance <= 0)] = np.nan
idx = np.nanargmin(distance, axis=1)
tf = events[idx]  # stimuli onset of the preceding stimulus
delta = 200  # 200 samples to simplify mock data
t0 = tf - delta
baseline = np.zeros((data_rp.shape[0], data_rp.shape[1], delta))
eegs = _picks_to_idx(info, "eeg", exclude=())
assert eegs.size == data_rp.shape[1]  # sanity-check
for k, (t1, t2) in enumerate(zip(t0, tf, strict=True)):
    baseline[k, :] = data[eegs, t1:t2]
baseline = np.mean(baseline, axis=-1, keepdims=True)  # average in time
data_rp_after = deepcopy(data_rp) - baseline

# viz
f, ax = plt.subplots(2, 2, figsize=(10, 10), sharex=True, sharey=True)
for n_epo in range(2):
    for n_ch in range(2):
        ax[n_epo, n_ch].plot(data_rp[n_epo, n_ch, :], label="before")
        ax[n_epo, n_ch].plot(data_rp_after[n_epo, n_ch, :], label="after", color="teal")
ax[0, 0].set_title("EEG1")
ax[0, 0].set_ylabel("Epoch 1")
ax[0, 1].set_title("EEG2")
ax[1, 0].set_ylabel("Epoch 2")
plt.show()
```

### CODE REVIEW

File: eeg_flow/tasks/epochs_evoked.py [hunk 1]
```diff
@@ -256,9 +262,32 @@ def _create_epochs_evoked_and_behavioral_metadata(
...
+        idx = np.nanargmin(distance, axis=1)
+        tf = events[idx, 0]  # stimuli onset of the preceding stimulus
+        delta = int(np.round(raw.info["sfreq"] * 0.2))  # 200 ms before stimulus onset
+        t0 = tf - delta
+        baseline = np.zeros(
+            (epochs_response._data.shape[0], epochs_response._data.shape[1], delta)
+        )
+        eegs = _picks_to_idx(raw.info, "eeg", exclude=())
+        assert eegs.size == epochs_response._data.shape[1]  # sanity-check
+        for k, (t1, t2) in enumerate(zip(t0, tf, strict=True)):
```
Comment (ArthurNguyen) [inline eeg_flow/tasks/epochs_evoked.py:286]: Hi there, something seems to be wrong here
I ran this on my ~84 (4 files for each of the 21p) files and only half work, 

To be precise it was able to output
19 files for oddball 1, 
19 for odball 2
2 for UT 1
5 for UT 2

Here's the error message, and I can send you a file if useful at this stage?

`
[3675525708:create_epochs_evoked_and_behavioral_metadata:190] ERROR: The file for participant P02, group G2, task UT, run 1 could not be processed.
[3675525708:create_epochs_evoked_and_behavioral_metadata:198] ERROR: could not broadcast input array from shape (64,0) into shape (64,205)
Traceback (most recent call last):
  File "C:\Users\anguyen\AppData\Local\Temp\ipykernel_48052\3675525708.py", line 112, in create_epochs_evoked_and_behavioral_metadata
    ) = _create_epochs_evoked_and_behavioral_metadata(raw)
  File "C:\Users\anguyen\AppData\Local\Temp\ipykernel_48052\3675525708.py", line 290, in _create_epochs_evoked_and_behavioral_metadata
    baseline[k, :] = raw[eegs, t1:t2][0]
ValueError: could not broadcast input array from shape (64,0) into shape (64,205)`

Any ideas?
Thanks!

Comment (mscheltienne) [reply eeg_flow/tasks/epochs_evoked.py:286]: Can you share one of the broken files? I need to drop into a debugger console to figure out why `raw[eegs, t1:t2][0]` yields an empty array.

Comment (ArthurNguyen) [reply eeg_flow/tasks/epochs_evoked.py:286]: here!
https://we.tl/t-5BLFo8HRT6

Comment (mscheltienne) [reply eeg_flow/tasks/epochs_evoked.py:286]: I forgot that we cropped the `raw` thus `raw.first_samp` is not equal to 0.

**Pull request merged**